src/App.tsx

```diff
@@ -1,12 +1,66 @@
 import React from 'react';
 import './App.scss';
+import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import Loader from 'react-loader-spinner';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+import { GoodsList } from './components/GoodsList';
+import { LoadingError } from './components/LoadingError';
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+interface State {
+  goods: Good[],
+  isLoading: boolean,
+  hasLoadingError: boolean,
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [],
+    isLoading: false,
+    hasLoadingError: false,
+  };
+
+  loadData = async (getData: () => Promise<Good[]>) => {
+    this.setState({
+      isLoading: true,
+    });
+
+    try {
+      const goods = await getData();
+
+      this.setState({
+        goods,
+      });
+    } catch {
+      this.setState({
+        hasLoadingError: true,
+      });
+    } finally {
+      this.setState({
+        isLoading: false,
+      });
+    }
+  };
```

Comment:
Error message will be rendered forever after the first error, you should reset `hasLoadingError` at the start of this method, like you do with `isLoading`